src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -885,13 +885,10 @@ public AbstractStringBuilder append(long l) {
      * @return  a reference to this object.
      */
     public AbstractStringBuilder append(float f) {
-        try {
-            FloatToDecimal.appendTo(f, this);
-        } catch (IOException e) {
-            throw new AssertionError(e);
-        }
+        ensureCapacityInternal(count + FloatToDecimal.MAX_CHARS);
```

Comment:
The maximum length of float to decimal is 15, and the maximum length of double to decimal is 24. The original exception handling was because the Appendable#append(char) method threw an IOException, which is no longer necessary.